cabal-install/Distribution/Solver/Modular/Explore.hs

```diff
@@ -60,7 +60,7 @@ backjump (EnableBackjumping enableBj) var initial xs =
           | otherwise                            = f (csAcc `CS.union` cs) cm'
 
     logBackjump :: ConflictSet QPN -> ConflictMap -> ConflictSetLog a
-    logBackjump cs cm = failWith (Failure cs Backjump) (cs, updateCM initial cm)
+    logBackjump cs cm = cm `seq` failWith (Failure cs Backjump) (cs, updateCM initial cm)
```

Comment:
I'd consider a bang pattern instead of `seq` for readability.

Also, in `exploreLog` below the `cm` is only updated when conflict counting is enabled, while it is always updated here. As you suggested in #3570, I think the `if countConflicts…` would make sense here as well. (Or, if you think that `seq`ing `cm` is already sufficient, remove the `if countConflicts` in `exploreLog` as well, which would IMHO improve readability).